conans/client/api/conan_api.py

```diff
@@ -239,39 +239,47 @@ def search_remote_recipes(self, query, remote):
         data["results"] = results
         return data
 
-    @api_method
-    def get_remote_recipe_revisions(self, reference, remote):
-        ref = ConanFileReference.loads(reference)
+    def _get_revisions(self, reference, ref_type, remote=None):
+        if ref_type == "package":
+            ref = PackageReference.loads(reference)
+        else:
+            ref = ConanFileReference.loads(reference)
+
+        # Class method name to get the specific revisions
+        method_name = f"get_{ref_type}_revisions"
         if ref.revision:
-            raise ConanException("Cannot list the revisions of a specific recipe revision")
-        if not remote:
-            raise ConanException("Must get a remote")
+            raise ConanException(f"Cannot list the revisions of a specific {ref_type} revision")
 
-        try:
-            return self.app.remote_manager.get_recipe_revisions(ref, remote=remote)
-        except NotFoundException:
-            # This exception must be catched manually due to a server inconsistency:
-            # Artifactory API returns an empty result if the recipe doesn't exist, but
-            # Conan Server returns a 404. This probably should be fixed server side,
-            # but in the meantime we must handle it here
-            return []
+        # Let's get all the revisions from a remote server
+        if remote:
+            try:
+                return getattr(self.app.remote_manager, method_name)(ref, remote=remote)
+            except NotFoundException:
+                # This exception must be catched manually due to a server inconsistency:
+                # Artifactory API returns an empty result if the recipe doesn't exist, but
+                # Conan Server returns a 404. This probably should be fixed server side,
+                # but in the meantime we must handle it here
+                return []
+        else:
+            # Let's get the revisions from the local cache
+            revs = getattr(self.app.cache, method_name)(ref)
+            results = []
+            for revision in revs:
+                timestamp = self.app.cache.get_timestamp(revision)
+                result = {
+                    "revision": revision.revision,
+                    "time": from_timestamp_to_iso8601(timestamp)
+                }
+                results.append(result)
+            return results
 
     @api_method
-    def get_local_recipe_revisions(self, reference):
-        ref = ConanFileReference.loads(reference)
-        if ref.revision:
-            raise ConanException("Cannot list the revisions of a specific recipe revision")
+    def get_package_revisions(self, reference, remote=None):
+        return self._get_revisions(reference, 'package', remote=remote)
```

Comment:
Following @memsharded [proposal](https://github.com/conan-io/conan/pull/9288/files#r675644552), this should become:

```python
    @api_method
    def get_package_revisions(self, reference, remote=None):
        reference = PackageReference.loads(reference)
        return self._get_revisions(reference, remote=remote)
```